src/experimental/ProbabilisticGraphicalModels/bayesian_network.jl

```diff
@@ -40,6 +44,76 @@ function BayesianNetwork{V}() where {V}
     )
 end
 
+"""
+    translate_BUGSGraph_to_BayesianNetwork(g::MetaGraph; init=Dict{Symbol,Any}())
+
+Translates a BUGSGraph (with node metadata stored in NodeInfo) into a BayesianNetwork.
+"""
+function translate_BUGSGraph_to_BayesianNetwork(g::MetaGraph)
+    # Retrieve variable labels (stored as VarNames) from g.
+    varnames = collect(labels(g))
+    n = length(varnames)
+    original_graph = g.graph
+
+    # Preallocate arrays/dictionaries.
+    names = Vector{VarName}(undef, n)
+    names_to_ids = Dict{VarName,Int}()
+    values = Dict{VarName,Any}()
+    distributions = Vector{Distribution}(undef, n)
+    deterministic_fns = Vector{Any}(undef, n)
+    stochastic_ids = Int[]
+    deterministic_ids = Int[]
+    is_stochastic = falses(n)
+    is_observed = falses(n)
+    node_types = Vector{Symbol}(undef, n)
+
+    for (i, varname) in enumerate(varnames)
+        symbol_name = varname
+        nodeinfo = g[varname]
+        names[i] = symbol_name
+        names_to_ids[symbol_name] = i
+        is_stochastic[i] = nodeinfo.is_stochastic
+        is_observed[i] = nodeinfo.is_observed
+
+        if nodeinfo.is_stochastic
+            distributions[i] = nodeinfo.node_function((), ())
```

Comment:
change the definition of `BayesianNetwork` just make `distributions` a vector of `Function` and store that instead of concrete `Distribution` object.